src/NLog/Targets/WebServiceTarget.cs

```diff
@@ -312,135 +335,304 @@ private Uri BuildWebServiceUrl(object[] parameterValues)
+        private abstract class HttpPostFormatterBase
         {
-            string soapAction;
-            if (this.Namespace.EndsWith("/", StringComparison.Ordinal))
+            protected HttpPostFormatterBase(WebServiceTarget target)
             {
-                soapAction = this.Namespace + this.MethodName;
+                Target = target;
             }
-            else
+
+            protected abstract string ContentType { get; }
+            protected WebServiceTarget Target { get; private set; }
+
+            public MemoryStream PrepareRequest(HttpWebRequest request, object[] parameterValues)
             {
-                soapAction = this.Namespace + "/" + this.MethodName;
+                InitRequest(request);
+
+                var ms = new MemoryStream();
+                WriteContent(ms, parameterValues);
+                return ms;
             }
-            request.Headers["SOAPAction"] = soapAction;
 
-            return PrepareSoapRequestPost(request, parameterValues, SoapEnvelopeNamespace, "soap");
+            protected virtual void InitRequest(HttpWebRequest request)
+            {
+                request.Method = "POST";
+                request.ContentType = string.Format("{1}; charset={0}", Target.Encoding.WebName, ContentType);
+            }
 
+            protected abstract void WriteContent(MemoryStream ms, object[] parameterValues);
         }
 
-        private MemoryStream PrepareSoap12Request(HttpWebRequest request, object[] parameterValues)
+        private class HttpPostFormEncodedFormatter : HttpPostTextFormatterBase
         {
-            return PrepareSoapRequestPost(request, parameterValues, Soap12EnvelopeNamespace, "soap12");
+            UrlHelper.EscapeEncodingFlag encodingFlags;
+
+            public HttpPostFormEncodedFormatter(WebServiceTarget target) : base(target)
+            {
+                encodingFlags = UrlHelper.GetUriStringEncodingFlags(target.EscapeDataNLogLegacy, true, target.EscapeDataRfc3986);
+            }
+
+            protected override string ContentType
+            {
+                get { return "application/x-www-form-urlencoded"; }
+            }
+
+            protected override string Separator
+            {
+                get { return "&"; }
+            }
+
+            protected override string GetFormattedContent(string parametersContent)
+            {
+                return parametersContent;
+            }
+
+            protected override string GetFormattedParameter(MethodCallParameter parameter, object value)
+            {
+                string parameterValue = Convert.ToString(value, CultureInfo.InvariantCulture);
+                string valueString = string.Empty;
+                if (string.IsNullOrEmpty(parameterValue))
+                {
+                    return string.Concat(parameter.Name, "=");
+                }
+                else
+                {
+                    var sb = new StringBuilder(parameterValue.Length + 20);
```

Comment:
Consider changing it into:

var sb = new StringBuilder(parameter.Name.Length + parameterValue.Length + 20);